app/src/main/java/com/github/se/eventradar/ui/viewProfile/ProfileUi.kt

```diff
@@ -113,68 +161,183 @@ fun ProfileUi(
+                                    ))
+                        Spacer(
+                            modifier = Modifier.width(10.dp).testTag("phoneNumberBirthDateSpacer"))
+                        OutlinedTextField(
+                            value = uiState.birthDate,
+                            onValueChange = viewModel::onBirthDateChanged,
+                            modifier = Modifier.weight(1f).testTag("birthDateTextField"),
+                            label = { Text("Birth Date") },
+                            isError = uiState.birthDateIsError,
+                            colors =
+                                OutlinedTextFieldDefaults.colors()
+                                    .copy(
+                                        focusedPrefixColor = MaterialTheme.colorScheme.primary,
+                                        unfocusedLabelColor = MaterialTheme.colorScheme.primary,
+                                    ))
+                      }
+                }
+
+                Button(
+                    onClick = {
+                      if (viewModel.validateFields()) {
+                        viewModel.updateUserInfo()
+                        isEditMode.value = false
+                      }
+                    },
+                    modifier = Modifier.padding(top = 10.dp).testTag("saveButton")) {
+                      Text("Save")
+                    }
+              } else {
+                NameText(
+                    name = "${uiState.firstName} ${uiState.lastName}",
+                    modifier = Modifier.testTag("name"))
+                StandardProfileInformationText(
+                    text = "@${uiState.username}", modifier = Modifier.testTag("username"))
+                Column(
+                    modifier = Modifier.fillMaxWidth().testTag("leftAlignedViewProfileColumn"),
+                ) {
+                  Text(
+                      text = "Bio",
+                      modifier = Modifier.padding(top = 16.dp).testTag("bioLabelText"),
+                      color = MaterialTheme.colorScheme.onSurfaceVariant,
+                      fontSize = 12.sp,
+                      style = MaterialTheme.typography.labelMedium)
+                  Text(
+                      text = uiState.bio,
+                      modifier = Modifier.padding(top = 4.dp).testTag("bioInfoText"),
+                      color = MaterialTheme.colorScheme.onSurface,
+                      fontSize = 16.sp,
+                      style = MaterialTheme.typography.bodyMedium)
+                  if (!isPublicView) {
+                    Row(
+                        modifier = Modifier.padding(top = 16.dp).testTag("phoneNumberBirthDateRow"),
+                        horizontalArrangement = Arrangement.SpaceBetween) {
+                          Column(modifier = Modifier.weight(1f).testTag("phoneNumberColumn")) {
+                            Text(
+                                text = "Phone Number",
```

Comment:
nit: maybe a nice feature to implement would be to include the country code but it might take too long, so not a priority